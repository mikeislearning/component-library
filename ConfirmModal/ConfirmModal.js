import React from 'react';
import { Modal, Text, View } from 'react-native';
import { PropTypes } from 'prop-types';
import { Button, Spacer } from 'app/components';

import styles from './ConfirmModal.styles';

export default class ConfirmModal extends React.Component {
  static propTypes = {
    cancelButton: PropTypes.shape({
      onPress: PropTypes.func.isRequired,
      text: PropTypes.string.isRequired,
    }),
    confirmButton: PropTypes.shape({
      onPress: PropTypes.func.isRequired,
      text: PropTypes.string.isRequired,
    }).isRequired,
    description: PropTypes.string,
    isVisible: PropTypes.bool,
    text: PropTypes.string.isRequired,
  };

  static defaultProps = {
    isVisible: false,
  };

  render() {
    const { cancelButton, confirmButton, description, isVisible, text } = this.props;

    if (!isVisible) {
      return null;
    }

    return (
      <Modal visible={isVisible} transparent={true}>
        <View style={styles.modalContainer}>
          <View style={styles.container}>
            <Text style={styles.text}>{text}</Text>
            {description && <Text style={[styles.text, styles.description]}>{description}</Text>}
            <View style={styles.buttonsContainer}>
              <Button
                accessibilityLabel={confirmButton.text}
                fullWidth
                onPress={confirmButton.onPress}
                textStyle={styles.button}
              >
                {confirmButton.text}
              </Button>
              <Spacer size="xSmall" />
              {cancelButton && (
                <Button
                  accessibilityLabel={cancelButton.text}
                  fullWidth
                  onPress={cancelButton.onPress}
                  textStyle={styles.button}
                  theme="transparent"
                >
                  {cancelButton.text}
                </Button>
              )}
            </View>
          </View>
        </View>
      </Modal>
    );
  }
}
