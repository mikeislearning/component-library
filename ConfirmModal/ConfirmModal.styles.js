import { StyleSheet } from 'react-native';

import { colors, fonts } from 'app/theme';

const styles = StyleSheet.create({
  button: {
    fontSize: 16,
    paddingVertical: 2,
  },
  buttonsContainer: {
    marginTop: 20,
    width: 245,
  },
  container: {
    alignItems: 'center',
    backgroundColor: colors.white,
    borderRadius: 8,
    marginHorizontal: 10,
    paddingBottom: 30,
    paddingHorizontal: 20,
    paddingTop: 40,
  },
  description: {
    fontSize: 15,
    marginTop: 20,
    textAlign: 'center',
  },
  modalContainer: {
    backgroundColor: colors.semiTransparent,
    flex: 1,
    justifyContent: 'center',
  },
  text: {
    color: colors.arriveBlack,
    fontFamily: fonts.family.messinaSans.regular,
    fontSize: fonts.size.large,
  },
});

export default styles;
