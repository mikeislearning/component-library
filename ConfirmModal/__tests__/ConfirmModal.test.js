import React from 'react';
import { render } from 'react-native-testing-library';

import ConfirmModal from '../ConfirmModal';

describe('<ConfirmModal />', () => {
  it('renders correctly', () => {
    const modal = render(
      <ConfirmModal
        cancelButton={{ onPress: () => {}, text: "Don't delete" }}
        confirmButton={{ onPress: () => {}, text: 'Yes, delete' }}
        isVisible={true}
        text="Are you sure?"
      />
    );
    expect(modal).toMatchSnapshot();
  });

  it('renders nothing if isVisible is false', () => {
    const modal = render(
      <ConfirmModal
        cancelButton={{ onPress: () => {}, text: "Don't delete" }}
        confirmButton={{ onPress: () => {}, text: 'Yes, delete' }}
        isVisible={false}
        text="Are you sure?"
      />
    );
    expect(modal).toMatchSnapshot();
  });
});
