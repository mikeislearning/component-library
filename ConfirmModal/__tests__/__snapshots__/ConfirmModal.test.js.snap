// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmModal /> renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.6)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 8,
          "marginHorizontal": 10,
          "paddingBottom": 30,
          "paddingHorizontal": 20,
          "paddingTop": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#272E39",
            "fontFamily": "MessinaSans-Regular",
            "fontSize": 22,
          }
        }
      >
        Are you sure?
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
            "width": 245,
          }
        }
      >
        <View
          accessibilityLabel="Yes, delete"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#4145C8",
              "borderColor": "#4145C8",
              "borderRadius": 6,
              "borderWidth": 1,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 15,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "MessinaSans-Regular",
                  "fontSize": 16,
                },
                Object {
                  "color": "#FFFFFF",
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 16,
                  "paddingVertical": 2,
                },
              ]
            }
          >
            Yes, delete
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "justifyContent": "center",
              },
              Object {
                "marginVertical": 5,
              },
            ]
          }
        />
        <View
          accessibilityLabel="Don't delete"
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0)",
              "borderRadius": 6,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 12,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "MessinaSans-Regular",
                  "fontSize": 16,
                },
                Object {
                  "color": "#726E67",
                },
                Object {
                  "fontSize": 16,
                  "paddingVertical": 2,
                },
              ]
            }
          >
            Don't delete
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`<ConfirmModal /> renders nothing if isVisible is false 1`] = `null`;
